### ISSUE TITLE
Provide trading parameters and the option to turn on and off 'Auto' buy and sell

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1168]
**Is your feature request related to a problem? Please describe.**
- It is patently difficult to make the Delivery and Trade mission beneficial to both parties
- Players should be allowed to select parameters and turn on and off the ***Auto*** buy and sell.

**Describe the rationale**
- How a trading and delivery mission is made possible should not be a *mystery* to players.

**Describe the goal(s) you have in mind**
- Trading parameters should reveal a bit more underlining work of how the ***Auto*** buy and sell work.

**Describe the solution you desire to see**
- Use checkboxes to allow a player to customize the type of trades allowed to be considered
- If the auto feature of trade is not selected, then, a decision panel would be presented for players to go through a series of questions to arbitrate if he agrees to a buy or a sell decision.

**Specifications**
- pre 3.7.0 build 9058

**Additional context**
- Based on some points in https://github.com/mars-sim/mars-sim/issues/732

**Comment by mokun** [comment]
@bevans2000 

Can we add a trade queue ? It will be a list of goods that the player would like to buy and sell.

**Comment by bevans2000** [comment]
> @bevans2000
> 
> Can we add a trade queue ? It will be a list of goods that the player would like to buy and sell.

Yes I think this is quite easy. The user would create the BuyList manually instead of automatic now.

**Comment by mokun** [comment]
We could even reuse of some elements of the UI from mission creation wizard for selecting what amount resources and parts to buy and sell.

**Comment by mokun** [comment]
I'd love to see a trade queue that displays wanted goods automatically generated by the settlement as well as goods manually added by players.

**Comment by mokun** [comment]
It seems on me that we may apply the trade queue idea to construction.

So we could make a construction queue  autogenerated for each settlement at the start of each sim.

 Players may keep it or change to it.
